addons/product/tests/test_variants.py

```diff
@@ -461,3 +461,46 @@ def test_07_create_no_create_dynamic_always(self):
         self.assertEqual(len(toto.attribute_line_ids.mapped('attribute_id')), 10)
         self.assertEqual(len(toto.attribute_line_ids.mapped('value_ids')), 100)
         self.assertEqual(len(toto.product_variant_ids), 0)
+
+class TestVariantsImages(common.TestProductCommon):
+
+    def setUp(self):
+        res = super(TestVariantsImages, self).setUp()
+
+        # To make the test independent of the demo data, you only need to rewrite this line
```

Comment:
Maybe check `test_website_sale_image.py` there might be all you need set up there to create your test without demo data.

Also I think there are overlaps with this, just the test about the one variant when other are archived is missing from there I believe.

Reply:
It was named `test_website_sale_zoom.py` in 12.0. It was less complete, I added a lot there in master actually.